### PR TITLE
Reorganize the casemapping crate, add more docs

### PR DESCRIPTION
**Pull request by Manishearth**
- `unfold` gets its own module
 - `CaseMappingInternals` is merged with `CaseMappingV1`. Data-related stuff is now in provider.rs, algorithms are still in internals.rs as a remote impl. The `internals` field in `CaseMapping` is similarly named
 - `exceptions_builder` is now under `provider`
 - Bunch of leftover docs and unstable banners

With this I am reasonably happy with the structure of this crate. Still need to take a closer look at the external API.

### CODE REVIEW

File: experimental/casemapping/src/casemapping.rs [hunk 1]
```diff
@@ -62,18 +62,11 @@ impl CaseMapping {
...
-        let internals = CaseMappingV1 {
-            casemap: CaseMappingInternals::try_from_icu(
-                trie_header,
-                trie_index,
-                trie_data,
-                exceptions,
-                unfold,
-            )?,
-        };
+        let data =
```
Comment (robertbastian) [inline experimental/casemapping/src/casemapping.rs:65]: Remove this method? Datagen doesn't need it, and it should not be in the rustdoc.

Comment (Manishearth) [reply experimental/casemapping/src/casemapping.rs:65]: yeah i was wondering why it exists.

**Review by robertbastian — APPROVED** [review]
Can you link an issue? I'd like to leave some todos

**Comment by Manishearth** [comment]
https://github.com/unicode-org/icu4x/issues/3234

**Pull request merged**